### ISSUE TITLE
Single file containing all modifiable parameters

### ISSUE THREAD

**Issue by rcooke-ast** [PYPIT/PYPIT#144]
As mentioned in the comments of PR #143, it would be very useful to have a single file that contains all available keyword arguments (i.e. settings) and their allowed list of parameters.

In principle, *all* available (modifiable) settings should be included in either the reduction or instrument settings files - this is a requirement of the new parser. One option might be to create a script that reads arparse.py, collects all of the available keywords arguments and the corresponding allowed values. The script should then output a documentation file containing all parameters and their description.

**Comment by profxj** [comment]
Am working in the *settings* branch on a new approach to 
reduce duplication amongst the various settings.instr files.

**Comment by rcooke-ast** [comment]
This functionality should already be present with the new parser. All that is needed is a file called `settings.instrument` which contains all keywords. Then, each instrument settings file should contain only the parameters that are different from the default values set in `settings.instrument`. This should only require 1/2 lines change in the code. I'm happy to take this on, if you want?

**Comment by profxj** [comment]
I'm already in it.  Take a peak at the *settings* branch:

https://github.com/PYPIT/PYPIT/compare/settings

and provide guidance on what I'm doing.  Thanks!

**Comment by rcooke-ast** [comment]
Sure thing!

First, I think I originally raised this issue as an intention to automatically generate documentation about the allowed settings/parameters PYPIT can read in. So, let's keep this issue open until this is created (something along the lines of a script that reads in arparse.py line by line and extractions information from the docstring and functions etc.).

Regarding the changes that need to be made, I think all that is needed is:
**(1)** Create a base argflag and spect settings file (as you have done) which contains all allowed parameters.
**(2)** Create an individual settings file for each pipeline or instrument, which only contains the parameters that deviate from the default values of the files created in step 1 (again, as you have done!).
**(3)** Modify code on line ~73 of `arparse.py`. I think you want a hook put in here that tells PYPIT which settings file to load. The first time this function is called, you want to load the base settings file (that you created in step 1 above), and the next time, you want to load the instrument/pipeline-specific settings files (that you created in step 2 above). I would do this with a additional argument for this function (something like `base=False`).
**(4)** Modify line ~133 of `pypit.py` to `lines = argf.load_file(base=True)`, and then modify line 137 to read:

```
    argf.set_paramlist(lines)
    lines = argf.load_file()
    argf.set_paramlist(lines)
```

So, the first time `load_file` is called (on line 133 or `pypit.py`), it will load the base argflag settings file, and then the second time `load_file` is called, it will overwrite the pipeline-specific settings. Similar modifications need to be made for lines 148/149 of `pypit.py` for the spectrograph settings.

I don't think the file: `pypit/data/settings/settings.py` is needed here.

**Comment by profxj** [comment]
Addressed by PR #400
